include/server.h

```diff
@@ -52,7 +53,7 @@ namespace kiwix
        void stop();
 
        void setRoot(const std::string& root);
-       void setAddress(const std::string& addr) { m_addr = addr; }
+       void setAddress(const std::vector<std::string>& addr) { m_addr = addr; }
```

Comment:
I am not sure that we can easily support such an API in general. To the best of my understanding you cannot set-up a single socket listening on (arbitrary) multiple addresses. A socket listening on *all* addresses is a special case. The commitment to such an API will require starting multiple servers (one per each address) which is doable but too much for the issue that we are trying to address.

Reply:
Sure, I will revert to taking a string as a parameter and initializing the vector on this.